packages/wp-story-editor/src/components/header/buttons/index.js

```diff
@@ -101,20 +105,24 @@ function Buttons() {
           <PreviewButton forceIsSaving={isSavingMetaBoxes} />
           {isSaving && <Loading />}
         </IconWithSpinner>
-        {isDraft ? (
+        {isPending && <SwitchToDraftButton forceIsSaving={isSavingMetaBoxes} />}
```

Comment:
Shouldn't the `SwitchToDraftButton` be just displayed only if `!isDraft`?

Currently, it's displayed when `isPending` or when `!isDraftOrPending`, it's a bit confusing -- that it's displayed if it's pending and if it's not pending. Sounds like just `!isDraft` would be sufficient.

Reply:
The difference is the order how the buttons are being rendered.

Draft stories get the following buttons, in this order:

1. Save as draft
2. Publish post (or submit for review if you're a contributor)

Published stories get:

1. Switch to draft
2. Update story 

Pending stories get:

1. Switch to draft
2. Update (if you're an admin; means continue saving as pending)
3. Publish post (or submit for review if you're a contributor)

Open for any suggestions to improve readability while maintaining the same order

Reply:
Hm, since now we have so many different statuses, maybe we should just display the buttons per status for readability, instead of trying to match the conditions? Even though some of those would repeat, it might still be worthwhile.

E.g.
```
 case 'pending`:
    return (... all buttons for pending ...);
 case `draft`:
    return (... all buttons for draft ...);
 ...etc.
```

(Preview button could be outside of the `switch` since that's always displayed for all statuses)
Thoughts?

Reply:
Improved now to make more readable 👍